orchestration/dagster_orchestration/hca_orchestration/contrib/retry.py

```diff
@@ -0,0 +1,40 @@
+import time
+from typing import Any, Callable
+
+
+class RetryException(Exception):
+    """
+    Represents a failure after some number of retries
+    """
+    pass
+
+
+def is_truthy(result: Any) -> bool:
+    """
+    Helper that returns True if the provided result is defined, False if not
+    """
+    return bool(result)
+
+
+def retry(target_fn: Callable[..., Any], max_wait_time_seconds: int,
+          poll_interval_seconds: int, success_predicate: Callable[[Any], bool], *args: str) -> Any:
```

Comment:
nit: i prefer keeping function definitions to either a single line or one line per argument, as follows:
```suggestion
def retry(
    target_fn: Callable[..., Any],
    max_wait_time_seconds: int,
    poll_interval_seconds: int,
    success_predicate: Callable[[Any], bool],
    *args: str,
) -> Any:
```

Reply:
Those should be `Any` I think, will make the change and thread through **kwargs as well.